File: Orion/ClientApp/src/app/shared/requests/product/add-product-price.request.ts
export class AddProductPriceRequest {
  productId: string;
  productUnitId: string;
  price: number;

  constructor(productId: string, productUnitId: string, price: number) {
    this.productId = productId;
    this.productUnitId = productUnitId;
    this.price = price;
  }

}

File: Orion/Controllers/Requests/Product/AddProductPriceRequest.cs
using FluentValidation;

namespace Orion.Controllers.Requests.Product
{
    public class AddProductPriceRequest
    {
        public Guid ProductId { get; set; }
        public Guid ProductUnitId { get; set; }
        public int Price { get; set; }
    }

    public class AddProductPriceRequestValidator : AbstractValidator<AddProductPriceRequest>
    {
        public AddProductPriceRequestValidator()
        {
            RuleFor(req => req.ProductId).NotEmpty();
            RuleFor(req => req.ProductUnitId).NotEmpty();
            RuleFor(req => req.Price).GreaterThanOrEqualTo(0).LessThanOrEqualTo(2147483647);
        }
    }
}
